src/solver/utils/optimization_statistics.h

```diff
@@ -0,0 +1,82 @@
+#ifndef __SOLVER_UTILS_STATISTICS_H__
+#define __SOLVER_UTILS_STATISTICS_H__
+
+#include <string>
+
+class optimizationStatistics
+{
+private:
+    long long totalSolveTime;
+    long long totalUpdateTime;
+    unsigned int nbSolve;
+    unsigned int nbUpdate;
+
+public:
+    optimizationStatistics() :
+        totalSolveTime(0),
+        totalUpdateTime(0),
+        nbSolve(0),
+        nbUpdate(0)
+    {
+    }
+
+    void add(const optimizationStatistics& other)
+    {
+        totalSolveTime += other.totalSolveTime;
+        totalUpdateTime += other.totalUpdateTime;
+        nbSolve += other.nbSolve;
+        nbUpdate += other.nbUpdate;
+    }
+
+    void addUpdateTime(long long updateTime)
+    {
+        totalUpdateTime += updateTime;
+        nbUpdate++;
+    }
+
+    void addSolveTime(long long solveTime)
+    {
+        totalSolveTime += solveTime;
+        nbSolve++;
+    }
+
+    unsigned int getNbUpdate() const
+    {
+        return nbUpdate;
+    }
+
+    unsigned int getNbSolve() const 
+    {
+        return nbSolve;
+    }
+
+    long long getTotalSolveTime() const
+    {
+        return totalSolveTime;
+    }
+
+    long long getTotalUpdateTime() const
+    {
+        return totalUpdateTime;
+    }
+
+    double getAverageUpdateTime() const
+    {
+        if (nbUpdate == 0) return 0.0;
+        return ((double) totalUpdateTime) / nbUpdate;
+    }
+
+    double getAverageSolveTime() const 
+    {
+        if (nbSolve == 0) return 0.0;
+        return ((double) totalSolveTime) / nbSolve;
+    }
+
+    std::string toString() const
+    {
+        return "average solve time: " + std::to_string(getAverageSolveTime()) + "ms, " 
+            + "average update time: " + std::to_string(getAverageUpdateTime()) + "ms";
+    }
```

Comment:
overloading std definitions may lead to undefined behavior https://en.cppreference.com/w/cpp/language/extending_std. It will probably work in practice but it seems dangerous. We can add to_string in another namespace or as a static method or keep toString()